sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledHashJoinExec.scala

```diff
@@ -321,17 +321,21 @@ object GpuShuffledHashJoinExec extends Logging {
       val startTime = System.nanoTime()
       var isBuildSerialized = false
       // Batches type detection
-      val coalesceBuiltIter = getHostShuffleCoalesceIterator(
-        bufBuildIter, buildDataType, targetSize, readOption, coalesceMetrics).map { iter =>
-        isBuildSerialized = true
+      val coalesceBuiltIter = getShuffleCoalesceIterator(
+        bufBuildIter, buildDataType, targetSize, readOption, coalesceMetrics).map {
+          case (iter, isHostResult) =>
+        isBuildSerialized = isHostResult
```

Comment:
could we change `isBuildSerialized` to `isHostSerialized` or something like that?